### PR TITLE
feat(portals-core): Navigation setup for portals

### PR DESCRIPTION
**Pull request by snaerseljan**
#   Navigation portals setup

This PR includes a setup for our for our portals navigation.
It includes, hooks, provider, filtering and more candy to use for our navigation.

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

File: apps/portals/admin/src/screens/Dashboard.tsx [hunk 1]
```diff
@@ -1,3 +1,10 @@
+import { IntroHeader } from '@island.is/portals/core'
+
 export const Dashboard = () => {
-  return <div>Dasboardinga</div>
+  return (
+    <IntroHeader
+      title="Stjórnborð Ísland.is"
+      intro="Hér eru stjórnborð sem þú hefur aðgang að fyrir ýmsar þjónustur tengdar Ísland.is."
```
Comment (disaerna) [inline apps/portals/admin/src/screens/Dashboard.tsx:7]: Should this be translated?

File: libs/portals/core/src/hooks/useNavigation.ts [hunk 2]
```diff
@@ -1,13 +1,75 @@
...
+  const routes = useRoutes()
+
+  const filteredNavigation = useMemo(() => {
+    if (userInfo) {
+      const nav = navigation || masterNav
+
+      return {
+        ...nav,
+        children: nav?.children?.filter((navItem) =>
+          filterNavigationTree(navItem, routes, userInfo, []),
```
Comment (saevarma) [inline libs/portals/core/src/hooks/useNavigation.ts:66]: ~Add a todo comment to handle dynamic routes when my-pages is migrated to use this.

**Review by saevarma — APPROVED** [review]
Really nice! Great job 💯

**Review by disaerna — APPROVED** [review]
LGTM

**Pull request merged**